### PR TITLE
[Search] Improve screenreader support

### PR DESCRIPTION
**Pull request by ncksllvn**
Goes with https://github.com/department-of-veterans-affairs/vagov-content/pull/410 (`search-breadcrumbs` branch of vagov-content)

For faster startup and reloads, start locally using `npm run watch -- --entry=search,static-pages`.

Ticket - https://github.com/department-of-veterans-affairs/vets.gov-team/issues/14813

---
## Description
This pull request modifies the Search app to use React-powered breadcrumbs instead of plain HTML so that browser focus can more easily be controlled.

- On the first render of the search results, browser focus should remain on the breadcrumbs.
- On subsequent result sets, browser focus should be applied to the heading for top recommendation, or the first link title.

Ticket - https://github.com/department-of-veterans-affairs/vets.gov-team/issues/14813

## Testing done
Local testing

## Screenshots
How are people recording those fancy GIFs?

## Acceptance criteria
- [x] Screenreader support is improved on the search app.

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/search/reducers/index.js [hunk 1]
```diff
@@ -41,6 +43,7 @@ function SearchReducer(state = initialState, action) {
         currentPage,
         perPage,
         totalPages,
+        searchesPerformed,
```
Comment (ncksllvn) [inline src/applications/search/reducers/index.js]: I added this counter into the reducer so we could easily tell when it's the first results set and therefore not to set browser focus.

File: src/applications/search/containers/SearchApp.jsx [hunk 2]
```diff
@@ -16,6 +18,10 @@ import IconSearch from '@department-of-veterans-affairs/formation-react/IconSear
 import Pagination from '@department-of-veterans-affairs/formation-react/Pagination';
 import AlertBox from '@department-of-veterans-affairs/formation-react/AlertBox';
 
+import SearchBreadcrumbs from '../components/SearchBreadcrumbs';
+
+const SCREENREADER_FOCUS_CLASSNAME = 'sr-focus';
```
Comment (ncksllvn) [inline src/applications/search/containers/SearchApp.jsx]: This class is applied to multiple elements. The helper `focusElement` will apply browser focus to the first one.

File: src/applications/search/containers/SearchApp.jsx [hunk 3]
```diff
@@ -51,13 +57,19 @@ class SearchApp extends React.Component {
...
-      this.writeBreadcrumb();
+    const hasNewResults =
+      prevProps.search.loading && !this.props.search.loading;
+
+    if (hasNewResults) {
+      const shouldFocusOnResults = this.props.search.searchesPerformed > 1;
+
+      if (shouldFocusOnResults) {
+        focusElement(`.${SCREENREADER_FOCUS_CLASSNAME}`);
+      }
```
Comment (ncksllvn) [inline src/applications/search/containers/SearchApp.jsx]: This reads as -

1. If the app was previously loading (`prevProps.search.loading`) and is not loading now (`!this.props.search.loading`), then we have new results
2. If we have new results, and this is not the first search, then we should focus on the result set
3. Focus!

**Review by 1Copenut — CHANGES_REQUESTED** [review]
@ncksllvn @aubreyarcangel 
I took a look at this on `localhost` with VoiceOver as my screenreader proxy, and the results were excellent. Things were taking focus as expected, and announcing the changed result list.

I have two visual change requests, and I'm thumbs up:

1. Can we add a rule to the `styles.scss` file to defeat the yellow focus halo on the breadcrumb and H4 for suggested results?
  ```css
    nav[aria-label="Breadcrumb"],
    h4.sr-focus {
      outline: 0;
    }
  ```
2. I think we'll need to set the `includeBreadcrumbs: false` YAML in **search.md** on the vagov-content repo to eliminate a double breadcrumb now that we're using React.

I'll review with JAWS and NVDA when we're in staging, but I expect them to pass as well. Thank you both for working through this issue. It's been great watching the team collaborate to solve a unique challenge.

**Comment by ncksllvn** [comment]
Thanks @1Copenut! Focus outline has been removed, and the Content PR to remove the breadcrumbs out of Markdown is here, https://github.com/department-of-veterans-affairs/vagov-content/pull/410.

@aubreyarcangel Can you take a look at this as well when you get the chance?

**Comment by ncksllvn** [comment]
@1Copenut @aubreyarcangel I'd like to merge this today, along with the [content PR](https://github.com/department-of-veterans-affairs/vagov-content/pull/410). Do you mind taking a look again when you have time?

**Review by 1Copenut — APPROVED** [review]

**Comment by 1Copenut** [comment]
@ncksllvn LGTM. I tested briefly on Chrome, Safari and Firefox on MacOS. Will test in Staging when ready.

**Comment by ncksllvn** [comment]
Thanks @1Copenut!

**Comment by 1Copenut** [comment]
@ncksllvn @aubreyarcangel I tested this in Staging with JAWS and NVDA. The results were very positive!

* **JAWS + IE11:** Announced top results header, number of results, breadcrumb change
* **JAWS + Chrome:** Announced top results header, search string change
* **NVDA + Firefox:** Announced top results header, partial search string change

How should we handle the open PR https://github.com/department-of-veterans-affairs/vets-website/pull/9743 ? Since we're handling the changes in React now, should that PR be closed without merging?

Thank you both for working through this on multiple iterations. I am thrilled to see how screen readers are behaving with search results now. I have some ideas for small results string readout improvements, and will put those in as an improvement ticket.

**Comment by ncksllvn** [comment]
I think PR https://github.com/department-of-veterans-affairs/vets-website/pull/9743 provided us with some great insight into how to structure our components to handle browser focus naturally, but I think it can be closed now. Thanks @aubreyarcangel for your work on that!

**Pull request merged**